v22.2/use-cloud-storage-for-bulk-operations.md

```diff
@@ -138,6 +138,115 @@ You [can associate an EC2 instance with an IAM role](https://docs.aws.amazon.com
+    - If you set the `Principal` ARN to `root`, this will allow any IAM user in the account with the `AssumeRole` permission to access the Amazon S3 bucket as per the defined IAM role permissions.
+    - When the IAM user takes on the role to perform a bulk operation, they have the permissions contained in the role. That is, not the permissions specified in their user profile.
+
+1. Run the bulk operation. It is necessary to pass in the S3 bucket's URL with the IAM user's access key ID and secret access key, if using [specified authentication](#specified-authentication). Otherwise, use `AUTH=IMPLICIT` for [implicit authentication](#implicit-authentication). For assuming the role, pass the assumed role's ARN, which you can copy from the IAM role's summary page:
+
+    ~~~sql
+    BACKUP DATABASE movr INTO 's3://{bucket name}?AWS_ACCESS_KEY_ID={user key}&AWS_SECRET_ACCESS_KEY={user secret key}&ASSUME_ROLE=arn:aws:iam::{account ID}:role/{role name}' AS OF SYSTEM TIME '-10s';
+    ~~~
+
+    CockroachDB also supports authentication for assuming roles when taking encrypted backups. To use with an encrypted backup, pass the `ASSUME_ROLE` parameter to the KMS URI as well as the bucket's: 
+
+    ~~~sql
+    BACKUP INTO 's3://{bucket name}?AWS_ACCESS_KEY_ID={user key}&AWS_SECRET_ACCESS_KEY={user secret key}&ASSUME_ROLE={ARN}'
+    WITH kms = 'aws:///{key}?AWS_ACCESS_KEY_ID={user key}&AWS_SECRET_ACCESS_KEY={user secret key}&REGION=us-east-1&ASSUME_ROLE={ARN}';
+    ~~~
+
+    For more information on the AWS KMS URI formats, see [Take and Restore Encrypted Backups](take-and-restore-encrypted-backups.html).
+
+#### Role chaining
+
+Beyond a user assuming a role, it is also possible to "chain" roles to create a path for users to assume roles to particular operations. Role chaining allows a user to assume a role through an intermediate role(s) instead of the user directly assuming a role. In this way, the role chain passes the request for access to the final role in the chain. Role chaining could be useful when a third-party organization needs access to your Amazon S3 bucket to complete a bulk operation. Or, your organization could grant roles based on limited-privilege levels.
+
+Assuming the role follows the same approach outlined in the previous section. The additional required step to chain roles is to ensure that the ARN of role A, which is assuming role B, is present in role B's trust policy with the `sts:AssumeRole` action. 
+
+The role B's trust policy must contain:
+
+~~~json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::{account-A-ID}:role/{role A name}"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+~~~
+
+In a chain of three roles, role C's trust policy needs to include role B in the same way. For example, to chain three roles so that a user could assume role C, it is necessary to verify the following:
```

Comment:
Nit: I would love to see a bolder notice that we move from "example chain of two roles" to "now we're thinking about a chain of three roles". Could be as simple as something like:

> Or, consider a chain of three roles, where role C's trust policy ...